en/docs/deploy/secure-userstore-using-pbkdf2.md

```diff
@@ -0,0 +1,87 @@
+# PBKDF2 hashing
+Password-Based Key Derivation Function 2 (PBKDF2) hashing algorithm is a modern hashing algorithm recommended by NIST. We can use the PBKDF2 hashing method to securely store user passwords in user stores. This method reduces the risk of brute-force attacks due to insecure passwords.
+
+This guide walks you through the steps of configuring PBKDF2 as the hashing algorithm of a JDBC userstore.
+
+!!! note
+    Currently, PBKDF2 supports only JDBC userstores of WSO2 Identity Server.
+
+## Configure PBKDF2 hashing
+To configure PBKDF2 hashing on a secondary JDBC user store:
+
+1. Login to the Identity Server management console (`https://<IS_HOST>:<PORT>/console`) and [create a JDBC user store](../configure-secondary-user-stores).
+
+    !!! info "Existing userstores"
+        - You may also use an existing user store which does not have any users in it. If you already have users in the userstore, once the hashing algorithm is configured these users will not be able to get authenticated.
+        - Such cases will impact with bad user experience as the users will not get authenticated even when they try to login using the correct credentials. Admins may use the following approaches to reset the user passwords after configuring the PBKDF2 hashing algorithm on an existing userstore:
+            - Ask users to [reset their own passwords](../guides/my-account/my-account.md).
+            - Trigger password reset for all accounts of the userstore using [admin initiated password reset](../guides/password-mgt/forced-password-reset.md).
+    
+    !!! info "PBKDF2 for primary JDBC userstores"
+        - PBKDF2 is also supported by [primary JDBC userstores](../configure-a-jdbc-user-store), but PBKDF2 **should be enabled before the initial server startup** by adding the following to the `deployment.toml` file.
+        ``` js
+        [user_store]
+        type = "database_unique_id"
+        password_digest="PBKDF2"
+        ```
```

Comment:
shall we have this as a separate section?
1 sub topic explains the secondary user store steps then the other topic will have the instructions to be followed for the primary user store

Reply:
Resolved

Added two new L3 heading under [L1] Configure PBKDF2 Hashing.
![image](https://user-images.githubusercontent.com/84217321/170926768-538195f6-afa0-4d3b-882b-82a20d379671.png)